fbchat/models.py

```diff
@@ -4,471 +4,169 @@
-        """Represents a Facebook thread"""
-        self.uid = str(uid)
-        self.type = _type
-        self.photo = photo
-        self.name = name
-        self.last_message_timestamp = last_message_timestamp
-        self.message_count = message_count
 
-    def __repr__(self):
-        return self.__unicode__()
+class FacebookError(Exception):
+    """Thrown by fbchat when Facebook returns an error
+
+    Attributes:
+        fb_error_code (int): The error code that Facebook returned
+        fb_error_message: A localized error message that Facebook returned
+    """
 
-    def __unicode__(self):
-        return '<{} {} ({})>'.format(self.type.name, self.name, self.uid)
+
+class Thread(object):
+    """Represents a Facebook chat-thread
+
+    Attributes:
+        id (int): The unique identifier of the thread
+        image: A url to the thread's thumbnail/profile picture
+        name: The name of the thread
+        last_message_timestamp: Timestamp of last message
+        message_count: Number of messages in the thread
+        nicknames: A dict, containing `User`\s and `Page`\s, mapped to their
+            nicknames
+        participants: Unique list of `User`\s and `Page`\s, denoting the
+            thread's participants
+        colour (`Colour`): The thread colour
+        color: Alias of :attr:`colour`
+        emoji: The thread's default emoji
+    """
 
 
 class User(Thread):
-    #: The profile url
-    url = None
-    #: The users first name
-    first_name = None
-    #: The users last name
-    last_name = None
-    #: Whether the user and the client are friends
-    is_friend = None
-    #: The user's gender
-    gender = None
-    #: From 0 to 1. How close the client is to the user
-    affinity = None
-    #: The user's nickname
-    nickname = None
-    #: The clients nickname, as seen by the user
-    own_nickname = None
-    #: A :class:`ThreadColor`. The message color
-    color = None
-    #: The default emoji
-    emoji = None
-
-    def __init__(self, uid, url=None, first_name=None, last_name=None, is_friend=None, gender=None, affinity=None, nickname=None, own_nickname=None, color=None, emoji=None, **kwargs):
-        """Represents a Facebook user. Inherits `Thread`"""
-        super(User, self).__init__(ThreadType.USER, uid, **kwargs)
-        self.url = url
-        self.first_name = first_name
-        self.last_name = last_name
-        self.is_friend = is_friend
-        self.gender = gender
-        self.affinity = affinity
-        self.nickname = nickname
-        self.own_nickname = own_nickname
-        self.color = color
-        self.emoji = emoji
+    """Represents a user and the chat-thread the client has with the user
+
+    Attributes:
+        first_name: The user's first name
+        last_name: The user's last name
+        is_friend (bool): Whether the user and the client are friends
+        gender: The user's gender
+        affinity (float): From 0 to 1. How close the client is to the user
```

Comment:
Maybe a few aliases to `Thread.nicknames`, such as `own_nickname` and `nickname`